### ISSUE TITLE
How can I calculate multiple objects of multiple class from tiny-yolo-voc object detection?

### ISSUE THREAD

**Issue by muradalqurishee10** [tensorpack/tensorpack#818]
I am using anaconda 3.6 for tiny YOLO for object detection. I am struggling to count the objects that are detected by yolo model in real time. I need to count multiple objects of multiple classes as shown in the image. Please give me a demo code. Thanks
The YOLO code is presented below
`tfnet = TFNet(options)
colors = [tuple(255 * np.random.rand(3)) for _ in range(10)]

capture = cv2.VideoCapture(0)
capture.set(cv2.CAP_PROP_FRAME_WIDTH, 1920)
capture.set(cv2.CAP_PROP_FRAME_HEIGHT, 1080)

while True:
    stime = time.time()
    ret, frame = capture.read()
    if ret:
        results = tfnet.return_predict(frame)
        for color, result in zip(colors, results):
            tl = (result['topleft']['x'], result['topleft']['y'])
            br = (result['bottomright']['x'], result['bottomright']['y'])
            X1= result['topleft']['x']
            Y1= result['topleft']['y']
            X2= result['bottomright']['x']
            Y2= result['bottomright']['y']
            area= (X2-X1)*(Y2-Y1)
            area_text = '{}: {:.0f}%'.format(area)
            label = result['label']
            confidence = result['confidence']
            text = '{}: {:.0f}%'.format(label, confidence * 100)
            frame = cv2.rectangle(frame, tl, br, color, 5)
            frame = cv2.putText(
                frame, text, area_text, tl, cv2.FONT_HERSHEY_COMPLEX, 1, (0, 0, 0), 2)
        cv2.imshow('frame', frame)
        print('FPS {:.1f}'.format(1 / (time.time() - stime)))
    if cv2.waitKey(1) & 0xFF == ord('q'):
        break

capture.release()
cv2.destroyAllWindows()`

**Comment by ppwwyyxx** [comment]
How is your question related to tensorpack?

**Comment by muradalqurishee10** [comment]
Yes. Keras Tensorflow backend

**Comment by ppwwyyxx** [comment]
How is "keras tensorflow backend" related to tensorpack?

**Comment by muradalqurishee10** [comment]
Yes, its related to tensorpack. I am using tensorflow.

Regards,


MURAD AL QURISHEE
Teaching Assistant  University of Tennessee at Chattanooga
Dept. of Civil and Chemical Engineering

m: (423)-316-8547 a: 862 Oak St, Chattanooga, TN 37403
w: http://muradalqurishee.wixsite.com/muradalqurishee
e: mgp216@ <muradalqurishee@gmail.com>mocs.utc.edu s: Murad Al Qurishee

Follow me

<https://newoldstamp.com/editor/?utm_source=Free_Editor&utm_medium=social_icon&utm_campaign=nos>
. <https://web.facebook.com/muradal.qurishee.7>.
<https://www.linkedin.com/in/muradal-qurishee>.

Get your own signature
<https://newoldstamp.com/editor/?utm_source=Free_Editor&utm_medium=email_banner&utm_campaign=new>

**Comment by ppwwyyxx** [comment]
"Using tensorflow" does not imply you're using tensorpack. Tensorflow is a dependency of tensorpack.

**Comment by muradalqurishee10** [comment]
Yes. I am using tensorpack.
Sorry for my lack of knowledge.

Regards,


MURAD AL QURISHEE
Teaching Assistant  University of Tennessee at Chattanooga
Dept. of Civil and Chemical Engineering

m: (423)-316-8547 a: 862 Oak St, Chattanooga, TN 37403
w: http://muradalqurishee.wixsite.com/muradalqurishee
e: mgp216@ <muradalqurishee@gmail.com>mocs.utc.edu s: Murad Al Qurishee

Follow me

<https://newoldstamp.com/editor/?utm_source=Free_Editor&utm_medium=social_icon&utm_campaign=nos>
. <https://web.facebook.com/muradal.qurishee.7>.
<https://www.linkedin.com/in/muradal-qurishee>.

Get your own signature
<https://newoldstamp.com/editor/?utm_source=Free_Editor&utm_medium=email_banner&utm_campaign=new>

**Comment by ppwwyyxx** [comment]
What functions or features in tensorpack are you having questions about?

**Comment by muradalqurishee10** [comment]
Function: Counter.

From my below code, How can I add a counter for count multiple class of multiple categories?
results = tfnet.return_predict(frame)
for color, result in zip(colors, results):
tl = (result['topleft']['x'], result['topleft']['y'])
br = (result['bottomright']['x'], result['bottomright']['y'])
label = result['label']

**Comment by ppwwyyxx** [comment]
There is no such function in tensorpack.
